opentelemetry-sdk/CHANGELOG.md

```diff
@@ -13,6 +13,7 @@
   `ProcessResourceDetector` resource detectors, use the
   [`opentelemetry-resource-detector`](https://crates.io/crates/opentelemetry-resource-detectors) instead.
 - Baggage propagation error will be reported to global error handler [#1640](https://github.com/open-telemetry/opentelemetry-rust/pull/1640)
+- Make `shutdown` method in `LoggerProvider` and `LogProcessor` taking immutable reference [#1643](https://github.com/open-telemetry/opentelemetry-rust/pull/1643)
```

Comment:
Lets describe all the changes this PR is making:
1. Loggers obtained after shutdown is now no-ops.
2. Shutdown requires immutable ref for provider/processor.
3. Simple and batch processors will ignore new logrecord after shutdown.